### DISCUSSION TITLE
WHy the color of grain boundary is not correct？

### DISCUSSION THREAD

**Discussion by Brian-Liu-rx** [mtex-toolbox/mtex#1027] [Ask Anything] [answered]
Hi，I set the color of grain boundary with misorientation > 15degree to be black,  [10 15] to be yellow,  [1 10] to be white.
However, like what shows below, the misorientation is calculated to be 16.3 degree, but the color is yellow, which is not right.

[grain_mtex,ebsd.grainId] = calcGrains(ebsd('indexed'),'threshold',1*degree);
plot(grain_mtex.boundary,'lineWidth',1,'lineColor',color_LAGB)
[grain_mtex,ebsd.grainId] = calcGrains(ebsd('indexed'),'threshold',10*degree);
plot(grain_mtex.boundary,'lineWidth',1,'lineColor',color_MAGB)
[grain_mtex,ebsd.grainId] = calcGrains(ebsd('indexed'),'threshold',15*degree);
plot(grain_mtex.boundary,'lineWidth',1,'lineColor',color_HAGB)

![QQ截图20210512104720](https://user-images.githubusercontent.com/69452222/117910958-7d1f0b80-b30f-11eb-9ccc-a184714a12d6.png)
![image](https://user-images.githubusercontent.com/69452222/117911066-adff4080-b30f-11eb-8902-c69693195d6e.png)

**Comment by Brian-Liu-rx** [comment]
color_LAGB = 'w';
color_MAGB = 'y';
color_HAGB = 'k';

**Comment by zmichels** [comment]
It looks like you are comparing the mean orientation from grains, whereas the  boundaries are based on the misorientation of neighboring orientations in the map. I would not expect the misorientation across a grain boundary to be identical to the misorientation between to grains mean-orientations.

Cheers,
Zach

**Reply by Brian-Liu-rx** [reply]
Thanks for the reply. Actually the orientation inside grain is uniform, because the EBSD original data is generated artificially，not from experiments

**Reply by Brian-Liu-rx** [reply]
The orientaion of pixels besides the grain boundary are also ori1 and ori2

**Comment by ralfHielscher** [comment, marked as answer]
most likely it is at 15 degree an inner boundary and therefore not plotted. You need to do:
```matlab
plot(grain_mtex.innerBoundary,'lineWidth',1,'lineColor',color_HAGB)
```

**Reply by Brian-Liu-rx** [reply]
Yes，It works after run this code:

plot(grain_mtex.innerBoundary,'lineWidth',1,'lineColor',color_HAGB)


Thanks!
